Sources/SwiftSyntaxMacroExpansion/MacroSystem.swift

```diff
@@ -715,14 +721,56 @@ private class MacroApplication<Context: MacroExpansionContext>: SyntaxRewriter {
       let visitedNode = self.visit(declSyntax)
       skipVisitAnyHandling.remove(Syntax(declSyntax))
 
-      let attributesToRemove = self.macroAttributes(attachedTo: visitedNode).map(\.attributeNode)
-
-      return AttributeRemover(removingWhere: { attributesToRemove.contains($0) }).rewrite(visitedNode)
+      let attributesToRemove = self.macroAttributes(attachedTo: visitedNode)
+      attributesToRemove.forEach { (attribute, spec) in
+        if let index = self.expandedAttributes.firstIndex(where: { expandedAttribute in
+          expandedAttribute.position == attribute.position
+        }) {
+          self.expandedAttributes.remove(at: index)
```

Comment:
Is removing necessary for correctness or do we just do it to keep the number of elements in `self.expandedAttributes` low (which is a good reason on its own). Would be worth a comment though, I think.

Reply:
It will be incorrect if it's possible that two attributes with same position are actually different during the process. For example, the list still contains an attribute expanded in the last visit, and in the same position there is attribute not expanded in this visit, I think it'll go wrong.

I think we should remove it from the list if it has been checked.